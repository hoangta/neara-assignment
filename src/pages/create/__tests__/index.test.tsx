import React from "react";
import { render, fireEvent } from '@testing-library/react';
import Create from '..'
import MFAStore from "../../../stores/mfaStore";

describe('Pages/Create', () => {
    const onBackSpy = jest.fn();
    const mockStore = new MFAStore();

    it('should render correctly', () => {
        const { asFragment } = render(<Create store={mockStore} onBack={onBackSpy} />);
        expect(asFragment()).toMatchSnapshot();
    })

    it('should create MFA and trigger onBack when click on Add button', () => {
        const { getByTestId } = render(<Create store={mockStore} onBack={onBackSpy} />);
        const testName = 'Test Name';
        const nameTextInput = getByTestId('name_text_input');
        fireEvent.change(nameTextInput, { target: { value: testName } });

        const addButton = getByTestId('add_button');
        fireEvent.click(addButton);
        expect(mockStore.createMFA).toBeCalledWith(testName);
        expect(onBackSpy).toBeCalled();
    })

    it('should disable Add button if name input is empty', () => {
        const { getByTestId } = render(<Create store={mockStore} onBack={onBackSpy} />);
        const nameTextInput = getByTestId('name_text_input');
        fireEvent.change(nameTextInput, { target: { value: '' } });

        const addButton = getByTestId('add_button');
        expect(addButton).toBeDisabled();
    })
})