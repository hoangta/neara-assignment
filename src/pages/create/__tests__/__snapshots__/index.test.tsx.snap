// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/Create should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="create_page"
  >
    <div
      class="container"
    >
      <button
        class="button"
        data-testid="back_button"
      >
        Back
      </button>
      <div
        class="title"
      >
        New token
      </div>
      <button
        class="button"
        data-testid="add_button"
        disabled=""
      >
        Add
      </button>
    </div>
    <input
      class="inputName"
      data-testid="name_text_input"
      maxlength="25"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
